### ISSUE TITLE
Feature: validate language-identifier format and duplicates in `require-meta-languages`

### ISSUE THREAD

**Issue by morgan-coded** [not-an-aardvark/eslint-plugin-eslint-plugin#645]
Following up on @bmish's review of #643: the `require-meta-languages` rule currently accepts any non-empty array of strings for `meta.languages` without inspecting the individual entries, so both of these pass today:

```js
// not a real language identifier
module.exports = { meta: { languages: ['javascript'] }, create(context) {} };

// duplicate entry
module.exports = { meta: { languages: ['js/js', 'js/js'] }, create(context) {} };
```

A future enhancement could:

- report entries that don't match the `pluginNamespace/languageName` identifier shape (e.g. `'js/js'`, `'json/json'`); and
- flag duplicate entries.

Tracking this separately from #643 as suggested. Since `meta.languages` is still fairly new and its semantics may evolve, it's worth settling how strictly to validate the identifier format — and against what source of truth — before implementing.

This would be a non-breaking addition to the existing (non-recommended) rule.

**Comment by bmish** [comment]
Let's actually just fix this now before we ship the rule:
- https://github.com/eslint-community/eslint-plugin-eslint-plugin/pull/643#discussion_r3591833640
